Incorporate roadmap into advanced sc intro/refesher notebook (#658)
AlexsLemonade/training-modules: scRNA-seq-advanced/01-read_filter_normalize_scRNA.Rmd

In this notebook, we will review basic processing for single-cell RNA-seq data, starting with the output from Cell Ranger, and proceeding through filtering, quality control, normalization, and dimension reduction. We will perform these tasks using tools from the [Bioconductor project](https://bioconductor.org), in particular [`SingleCellExperiment` objects](https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html) and functions that work with those objects.
Much of the material in this notebook is directly inspired by, and draws heavily on, material presented in the book [_Orchestrating Single Cell Analysis with Bioconductor_ (OSCA)](http://bioconductor.org/books/3.16/OSCA/).

> ![Single-cell roadmap overview](diagrams/roadmap_single_overview.png)

Reviewer: Generally for these "text captions," would a colon help?

```suggestion
![Single-cell roadmap: Overview](diagrams/roadmap_single_overview.png)
```

This will usually also include a bit of metadata about the features, including gene symbols (if the features are genes) and the type of data they represent (e.g., gene expression or antibody capture).
- `matrix.mtx.gz`, The counts themselves, stored in a sparse ["Matrix Exchange" format](https://math.nist.gov/MatrixMarket/formats.html).

Cell Ranger will also export these data in a single `HDF5` format file with a `.h5` extension, which can also be imported with the same commands we will use below.

> However, we have found that processing large `.h5` files is often _much_ less efficient in R, so so we prefer to start with the matrix files when possible.

Reviewer: ```suggestion
However, we have found that processing large `.h5` files is often _much_ less efficient in R, so we prefer to start with the matrix files when possible.
```

In particular, we would not recommend working with `.h5` files for raw data; the filtering steps we will use below can sometimes take hours when using those files as input.

> We will also need a table of mitochondrial genes, which we have stored in the `data/reference` directory. 

Reviewer: Well, I'm here anyways, so I might ask for a slash? I tend to like slashes to denote _directory_ vs file in a written path.
```suggestion
We will also need a table of mitochondrial genes, which we have stored in the `data/reference/` directory. 
```

## Reading Cell Ranger data

> ![Single-cell roadmap preprocessing](diagrams/roadmap_single_preprocess.png)

Reviewer: To match the diagram:

```suggestion
![Single-cell roadmap: Preprocess and Import](diagrams/roadmap_single_preprocess.png)
```

Whether the 10X Cell Ranger data is in Matrix Exchange format or in an HDF5 file, we can use the `read10xCounts()` function from the `DropletUtils` package to read the data into R and create a `SingleCellExperiment` object.

> (Though again, we do not recommend using the HDF5 file if you can avoid it, _especially_ for raw data.)

Reviewer: for consistency 
```suggestion
(Though again, we do not recommend using the `HDF5` file if you can avoid it, _especially_ for raw data.)
```

Also wondered if you might want to say "unfiltered" instead of "raw"? Or a parenthetical, `raw (unfiltered) data.`

Author: I used "raw" because that is the file/folder is called by Cell Ranger. I actually wrote it with "unfiltered," but then changed it to match Cell Ranger. Parenthetical seems good. 
```suggestion
(Though again, we do not recommend using the `.h5` file if you can avoid it, _especially_ for raw (unfiltered) data.)
```

As we proceed to calculate per-cell statistics, we will be adding new data to this table.

## Quality control and filtering

> ![Single-cell roadmap QC, filtering, and normalization](diagrams/roadmap_single_qc_norm.png)

Reviewer: ```suggestion
![Single-cell roadmap: QC, Filter, and Normalize](diagrams/roadmap_single_qc_norm.png)
```

## Dimension reduction

> ![Single-cell roadmap dimension reduction](diagrams/roadmap_single_dimension_reduction.png)

Reviewer: ```suggestion
![Single-cell roadmap: Dimension reduction](diagrams/roadmap_single_dimension_reduction.png)
```
